ml-agents/mlagents/trainers/cli_utils.py

```diff
@@ -189,6 +189,9 @@ def _create_parser() -> argparse.ArgumentParser:
         action=RaiseRemovedWarning,
         help="(Removed) Use the TensorFlow framework.",
     )
+    argparser.add_argument(
+        "--results-dir", default="results", help="Results base directory"
+    )
```

Comment:
Is this relevant to this PR? Why are we adding this flag ?

Reply:
This was previously hardcoded in learn.py
https://github.com/Unity-Technologies/ml-agents/pull/4788/files#diff-4e0d7b8aef419254e6fc3c63f5baa53f222afda7c908705e857a22f7d4792c47L68

and is now used in CheckpointSettings.maybe_init_path()
https://github.com/Unity-Technologies/ml-agents/pull/4788/files#diff-546e90789e914f8707fd97391f78c4bba39ae69a965858bbb69c2d324db1ec51R719

I can keep it hardcoded if you prefer, but I do think moving all the path logic to part of RunOptions is a win (and basically a pre-req to make TensorBoardWriter act like a plugin)

Reply:
If a user changes this from the default `results` will it mess with how they're saved in cloud? cc: @hvpeteet

Reply:
If a user can set it in the run_options then yes this could mess with cloud since we don't already take it into account. I don't think it is a reason not to make the change though. We can add logic to overwrite this field since we already modify run_options. 

Just a couple of questions to confirm my understanding:

1. Would this be set under `checkpoint_settings --> write_path`?
2. Could we blindly set this for every experiment and it would be ignored by older mlagents versions?

Reply:
1. It would be `checkpoint_settings -> results_dir` (at the top level)
2. This would trigger an error here https://github.com/Unity-Technologies/ml-agents/blob/5db6efebad7c27ea0437a37f038214a09d9e9d26/ml-agents/mlagents/trainers/settings.py#L37

Reply:
Thanks, we can check for it and only overwrite it if already set then. When you check this in can you file a ticket against me to make sure we get this added at some point?

Reply:
Will do